src/pulsar_utils.erl

```diff
@@ -84,3 +84,41 @@ proplist_diff(Opts1, Opts2) ->
             end,
             proplists:delete(Key, Opts1Acc)
         end, Opts1, Opts2).
+
+%% copied from `edoc_lib' because dialyzer cannot see this private
+%% function there.
+escape_uri([C | Cs]) when C >= $a, C =< $z ->
+    [C | escape_uri(Cs)];
+escape_uri([C | Cs]) when C >= $A, C =< $Z ->
+    [C | escape_uri(Cs)];
+escape_uri([C | Cs]) when C >= $0, C =< $9 ->
+    [C | escape_uri(Cs)];
+escape_uri([C = $. | Cs]) ->
+    [C | escape_uri(Cs)];
+escape_uri([C = $- | Cs]) ->
+    [C | escape_uri(Cs)];
+escape_uri([C = $_ | Cs]) ->
```

Comment:
Oops, sorry, missed that `_` is allowed, so there is going to be a clash.
eg.
the topic name `_2Froot`
will clash with the topic name `/root` after escape.
we must either escape `_`, or use a different symbol, e.g. `=`